agent/discovery/sapsystem.go

```diff
@@ -26,28 +28,39 @@ func (d SAPSystemsDiscovery) GetId() string {
 	return d.id
 }
 
-func (d SAPSystemsDiscovery) Discover() error {
+func (d SAPSystemsDiscovery) Discover() (string, error) {
+	var sids, plural string
 	systems, err := sapsystem.NewSAPSystemsList()
 
 	if err != nil {
-		return err
+		return fmt.Errorf("%v", err).Error(), err
 	}
 
 	d.SAPSystems = systems
 	for _, s := range d.SAPSystems {
+		if sids != "" {
```

Comment:
We already have the method to get this SID list.
https://github.com/trento-project/trento/blob/e70765b81566f41a1b0f2df86916e8f96a851bcd/internal/sapsystem/sapsystem.go#L103

You can simply is this at then end to create the output string.
I wouldn't be worried about the plural. You can simply put `SAP system(s)` maybe

Reply:
Changed!